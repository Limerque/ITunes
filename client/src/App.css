body {
  font-family: Arial, sans-serif;
}

h1,
h2,
h3 {
  margin-top: 0;
}

.container {
  max-width: 800px;
  margin: 0 auto;
  padding: 20px;
}

form {
  display: flex;
  margin-bottom: 20px;
}

input[type="text"] {
  flex-grow: 1;
  padding: 5px;
  margin-right: 10px;
}

select {
  padding: 5px;
  margin-right: 10px;
}

button {
  padding: 5px 10px;
  background-color: rgb(121, 21, 46);
  color: white;
  border: none;
  cursor: pointer;
}

.search-results-list {
  list-style-type: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
}

.search-result-item {
  width: 200px;
  margin-right: 20px;
  margin-bottom: 20px;
}

.result-image {
  width: 100%;
  height: auto;
  margin-bottom: 10px;
}

.result-details {
  text-align: center;
}

h3 {
  margin: 0;
  font-size: 14px;
}

p {
  margin: 0;
  font-size: 12px;
}

a {
  color: black;
  text-decoration: none;
  text-decoration: underline;
}

.favorites-list {
  list-style-type: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
}

.favorites-item {
  width: 200px;
  margin-right: 20px;
  margin-bottom: 20px;
}
.navbar {
  background-color: rgb(121, 21, 46);
  padding: 10px;
  display: flex;
  justify-content: center;
}

.navbar ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: center;
}

.navbar li {
  margin: 0 200px;
}

.navbar a {
  text-decoration: none;
  color: whitesmoke;
  font-weight: bold;
  padding: 5px;
}

.navbar a:hover {
  color: #000;
  background-color: rgb(121, 21, 46);
}
