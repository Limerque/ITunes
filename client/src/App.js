import React, { useState } from "react";
import { BrowserRouter as Router, Route, Link, Switch } from "react-router-dom";
import SearchForm from "./components/SearchForm";
import SearchResults from "./components/SearchResults";
import FavoritesPage from "./components/FavoritesPage";
import "./App.css";

function App() {
  const [term, setTerm] = useState(""); // State to store the search term
  const [mediaType, setMediaType] = useState("all"); // State to store the selected media type
  const [searchResults, setSearchResults] = useState([]); // State to store the search results
  const [favorites, setFavorites] = useState([]); // State to store the favorite items

  const handleSearch = async (e) => {
    e.preventDefault();
    try {
      const response = await fetch(
        `/api/search?term=${term}&mediaType=${mediaType}`
      );
      const data = await response.json();
      setSearchResults(data); // Update the search results state with the fetched data
    } catch (error) {
      console.error("Error fetching search results:", error);
    }
  };

  const handleAddToFavorites = (item) => {
    setFavorites((prevFavorites) => [...prevFavorites, item]);
    alert("Item added to favorites"); // Add an item to the favorites state
  };

  const handleRemoveFromFavorites = (item) => {
    setFavorites((prevFavorites) =>
      prevFavorites.filter((fav) => fav.trackId !== item.trackId)
    );
    alert("Item removed from favorites"); // Remove an item from the favorites state based on its trackId
  };

  return (
    <Router>
      <div>
        <nav className="navbar">
          <ul>
            <li>
              <Link to="/">Search</Link> {/* Link to the search page */}
            </li>
            <li>
              <Link to="/favorites">Favorites</Link>{" "}
              {/* Link to the favorites page */}
            </li>
          </ul>
        </nav>

        <Switch>
          <Route exact path="/">
            <div>
              <h1>iTunes Search</h1>
              <SearchForm
                term={term}
                mediaType={mediaType}
                onSearch={handleSearch}
                onTermChange={setTerm}
                onMediaTypeChange={setMediaType}
              />

              <SearchResults
                results={searchResults}
                onAddToFavorites={handleAddToFavorites}
              />
            </div>
          </Route>
          <Route path="/favorites">
            <FavoritesPage
              favorites={favorites}
              onRemoveFromFavorites={handleRemoveFromFavorites}
            />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default App;
