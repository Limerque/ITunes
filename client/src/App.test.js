import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import App from "./App";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <App />
      </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
