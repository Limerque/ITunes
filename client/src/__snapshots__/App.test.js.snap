// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Search
        </a>
         
      </li>
      <li>
        <a
          href="/favorites"
          onClick={[Function]}
        >
          Favorites
        </a>
         
      </li>
    </ul>
  </nav>
  <div>
    <h1>
      iTunes Search
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Search term"
        type="text"
        value=""
      />
      <select
        onChange={[Function]}
        value="all"
      >
        <option
          value="all"
        >
          All
        </option>
         
        <option
          value="movie"
        >
          Movie
        </option>
         
        <option
          value="podcast"
        >
          Podcast
        </option>
         
        <option
          value="music"
        >
          Music
        </option>
         
        <option
          value="audiobook"
        >
          Audiobook
        </option>
         
        <option
          value="shortFilm"
        >
          Short Film
        </option>
         
        <option
          value="tvShow"
        >
          TV Show
        </option>
         
        <option
          value="software"
        >
          Software
        </option>
         
        <option
          value="ebook"
        >
          eBook
        </option>
         
      </select>
      <button
        type="submit"
      >
        Search
      </button>
       
    </form>
    <div>
      <h2>
        Search Results
      </h2>
      <ul
        className="search-results-list"
      />
    </div>
  </div>
</div>
`;
